sdk/eventhubs/azure-messaging-eventhubs/src/main/java/com/azure/messaging/eventhubs/EventData.java

```diff
@@ -213,6 +212,22 @@ public Long getSequenceNumber() {
         return systemProperties.getSequenceNumber();
     }
 
+    /**
+     * Gets the sequence number that was assigned during publishing, if the event was successfully
+     * published by a sequence-aware producer.  If the producer was not configured to apply
+     * sequence numbering or if the event has not yet been successfully published, this member
+     * will be {@code null}.
+     *
+     * The published sequence number is only populated and relevant when certain features
+     * of the producer are enabled. For example, it is used by idempotent publishing.
+     *
+     * @return The publishing sequence number assigned to the event at the time it was successfully published.
+     * {@code null} if the {@link EventData} hasn't been sent or it's sent without idempotent publishing enabled.
+     */
+    public Integer getPublishedSequenceNumber() {
```

Comment:
This isn't set until publishing is confirmed as successful, correct? (we don't want to represent an interim state to callers when the publishing operation is active)

Reply:
As of your asking, it's set just before publishing. Changed to set it after publishing is successful.